ca/server.go

```diff
@@ -464,7 +467,12 @@ func (s *Server) Run(ctx context.Context) error {
 // Stop stops the CA and closes all grpc streams.
 func (s *Server) Stop() error {
 	s.mu.Lock()
+
+	// Wait for Run to complete before returning
+	defer s.wg.Wait()
```

Comment:
Also, `s.cancel()` seems to be called after the check for `!s.isRunning()` - would this just block unless something else cancelled the `Run` context?